src/surge-xt/osc/OpenSoundControl.cpp

```diff
@@ -112,34 +114,106 @@ void OpenSoundControl::oscMessageReceived(const juce::OSCMessage &message)
     std::string address1, address2, address3;
     std::getline(split, address1, '/');
 
-    if (address1 == "param")
+    if (address1 == "fnote")
+    // Play a note at the given frequency and velocity
+    {
+        std::getline(split, address2, '/'); // check for '/rel'
+
+        if (!message[0].isFloat32())
+        {
+            sendError("Invalid data type for frequency (must be a float).");
+            return;
+        }
+
+        if (!message[1].isFloat32())
+        {
+            sendError("Invalid data type for velocity (must be an float between 0 - 127).");
+            return;
+        }
+
+        float32_t frequency = message[0].getFloat32();
+        int velocity = static_cast<int>(message[1].getFloat32());
+
+        // ensure freq. is in MIDI note range
+        if (frequency < MIDI_MIN_FREQ || frequency > MIDI_MAX_FREQ)
```

Comment:
So here you could use that constant and the constant * 1<<7

Reply:
OK. Are the tunings constants in scope for OpenSoundControl.cpp?

Reply:
They seem to be